### PR TITLE
Convert documentation in docs/ to jupyter notebooks (new PR)

### PR DESCRIPTION
**Pull request by bt3gl**
New PR for files in PR #2542, fixing [#2310](https://github.com/quantumlib/Cirq/issues/2310).

- Remove `schedule` notebook (and edit texts to match to documentation). 

- Add a notebook for each of the [markdown pages inside docs/](https://github.com/quantumlib/Cirq/blob/master/docs/) (checked against `latest`). I also checked everything with Grammarly, fixing grammar typos when necessary:

Screen Shot 2020-01-27 at 5 00 16 PM
Screen Shot 2020-01-27 at 5 00 21 PM
Screen Shot 2020-01-27 at 5 00 28 PM
Screen Shot 2020-01-27 at 5 00 38 PM
Screen Shot 2020-01-27 at 5 00 42 PM
Screen Shot 2020-01-27 at 5 00 46 PM

### Testing Locally

Install [anaconda](https://www.anaconda.com/) and [miniconda](https://docs.conda.io/en/latest/miniconda.html), and create a virtual environment with:

```
conda create -n venv python=3
conda activate venv
conda install --yes --file requirements.txt
conda install -c conda-forge google-api-python-client
conda install -c conda-forge cirq
```

### CODE REVIEW

File: docs/notebooks/gates.ipynb [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Classes that implement ``Gate`` must supply either the ``_num_qubits_`` or ``_qid_shape_`` *magic methods*.  For convenience, one can use the ``SingleQubitGate``, ``TwoQubitGate``, and ``ThreeQubitGate`` classes for these common gate sizes.\n",
+    "\n",
+    "The most common type of ``Gate`` is one that corresponds to applying\n",
+    "a unitary evolution on the qubits that the gate acts on.\n",
+    "``Gate`` objects can also correspond to noisy evolution on the qubits. This version of a gate is not used when sending the circuit to a\n",
+    "quantum computer for execution, but it can be used with\n",
+    "various simulators. See [noise documentation](noise.md)."
```
Comment (vtomole) [inline docs/notebooks/gates.ipynb]: Link to Jupyter instead of md

File: docs/notebooks/gates.ipynb [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+    "A `Gate` must implement the `_num_qubits_` (or `_qid_shape_`) method.\n",
+    "This method returns an integer and is used by `cirq.num_qubits` to determine how many qubits this gate operates on.\n",
+    "\n",
+    "#### `cirq.qid_shape` and `def _qid_shape_`\n",
+    "\n",
+    "A qudit gate or operation must implement the `_qid_shape_` method that returns a tuple of integers.\n",
+    "This method is used to determine how many qudits the gate or operation operates on and what dimension each qudit must be.\n",
+    "If only the `_num_qubits_` method is implemented, the object is assumed to operate only on qubits.\n",
+    "Callers can query the qid shape of the object by calling `cirq.qid_shape` on it.\n",
+    "See [qudit documentation](qudits.md) for more information.\n",
```
Comment (vtomole) [inline docs/notebooks/gates.ipynb]: Link to Jupyter instead of md

File: docs/notebooks/simulation.ipynb [hunk 3]
```diff
@@ -0,0 +1,626 @@
...
+    "In addition, the ``StepResult`` contains ``set_state()``,\n",
+    "which  can be used to set the ``state``. One can pass a valid\n",
+    "full state to this method by passing a numpy array. Or,\n",
+    "alternatively, one can pass an integer, and then the state\n",
+    "will be set lie entirely in the computation basis state\n",
+    "for the binary expansion of the passed integer.\n",
+    "\n",
+    "### Monte Carlo simulations of noise\n",
+    "\n",
+    "Some [noise](noise.md) models can be thought of as randomly\n",
```
Comment (vtomole) [inline docs/notebooks/simulation.ipynb]: Link to Jupyter instead of md

File: docs/notebooks/simulation.ipynb [hunk 4]
```diff
@@ -0,0 +1,626 @@
...
+    "`SupportsMixture` protocol.  Also, note that this functionality\n",
+    "is currently only supported in the pure state simulator and\n",
+    "not in the density state simulator.  If the mixed state simulator\n",
+    "encounters a mixture, it will treat it as a general channel.\n",
+    "\n",
+    "### Parameterized Values and Studies\n",
+    "\n",
+    "In addition to circuit gates with fixed values, Cirq also\n",
+    "supports gates which can have ``Symbol`` value (see\n",
+    "[Gates](gates.md)). These are values that can be resolved\n",
```
Comment (vtomole) [inline docs/notebooks/simulation.ipynb]: Link to Jupyter instead of md

File: docs/notebooks/circuits.ipynb [hunk 5]
```diff
@@ -0,0 +1,755 @@
...
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Circuits\n",
+    "\n",
+    "## Conceptual Overview\n",
+    "\n",
+    "There are two primary representations of quantum programs in Cirq, each of which are represented by a class: `Circuit` and `Schedule`. \n",
```
Comment (vtomole) [inline docs/notebooks/circuits.ipynb]: Delete schedule

File: docs/notebooks/circuits.ipynb [hunk 6]
```diff
@@ -0,0 +1,755 @@
...
+    "\n",
+    "Conceptually, a `Circuit` object is very closely related to the abstract quantum circuit model, while a `Schedule object` is like the abstract quantum circuit model but includes detailed timing information.\n",
```
Comment (vtomole) [inline docs/notebooks/circuits.ipynb]: Delete schedule

File: docs/notebooks/tutorial.ipynb [hunk 7]
```diff
@@ -0,0 +1,890 @@
...
+    "The above shows that one can use a quantum computer to obtain estimates of the objective function for the ansatz. This can then be used in an outer loop to try to obtain parameters for the lowest value of the objective function. For these values, one can then use that best ansatz to produce samples of solutions to the problem, which obtain a hopefully good approximation for the lowest possible value of the objective function.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create a circuit on a Grid\n",
+    "\n",
+    "To build the above variational quantum algorithm using Cirq, one begins by building the appropriate circuit. In Cirq, circuits are represented either by a `Circuit` object or a `Schedule` object. \n",
```
Comment (vtomole) [inline docs/notebooks/tutorial.ipynb]: delete schedule.

File: docs/notebooks/tutorial.ipynb [hunk 8]
```diff
@@ -0,0 +1,890 @@
...
+    "\n",
+    "`Schedule` objects offer more control over quantum gates and circuits at the timing level, which we do not need, so here we will work with `Circuit` objects instead.\n",
```
Comment (vtomole) [inline docs/notebooks/tutorial.ipynb]: Delete schedule.

Comment (bt3gl) [reply docs/notebooks/tutorial.ipynb]: Done: I replaced everything by the new (`.md`) texts in https://cirq.readthedocs.io/en/stable/tutorial.html

**Comment by vtomole** [comment]
@mpharrigan I haven't looked into `nbsphinx`. The next issue after this is to make a check that executes these notebooks the way `snippets_test.py` does. Maybe `nbsphinx` can do that.  Can it convert `.md` to jupyter notebooks? That would make our job easy.

**Comment by mpharrigan** [comment]
The idea would be to dump the `.md` files and have `nbsphinx` execute and render the notebooks into the final sphinx html output. I spot checked it with the new `circuits.ipynb` and it worked out of the box and looked pretty good but someone should do it with a more exhaustive test for regressions

**Pull request merged**